message/messaging_service.cc

```diff
@@ -660,6 +660,14 @@ messaging_service::scheduling_group_for_isolation_cookie(const sstring& isolatio
             return info.sched_group;
         }
     }
+    // Check for the case of the client using a connection class we don't
+    // recognize, but we know its a tenant, not a system connection.
+    // Fall-back to the default tenant in this case.
+    for (auto&& connection_prefix : _connection_types_prefix) {
+        if (isolation_cookie.find(connection_prefix.data()) == 0) {
```

Comment:
Why not isolation_cookie == connection_prefix?

The find() thing needs deciphering. It's not that complicated, but "==" is much simper, no?

Reply:
I doesn't work. `isolation_cookie` will be something like `statement:default` for the case this patch wants to solve.

Reply:
Yes, it even says "prefix".